pkg/minishift/openshift/service.go

```diff
@@ -238,6 +220,37 @@ func filterAndUpdateServiceSpecs(routeSpec map[string][]ServiceWeight, namespace
 	return serviceSpecs
 }
 
+func addStuffToServiceSpecMap(serviceSpecMap map[string]*ServiceSpec, url string, service ServiceWeight, namespace string, np string) {
+	const (
+		http  = "http://%s"
+		https = "https://%s"
+	)
+
+	if _, ok := serviceSpecMap[service.Name]; !ok {
+		if service.Tls != "" {
+			serviceSpecMap[service.Name] = fillServiceSpec(service.Name, url, np, service.Weight, namespace, https)
+		} else {
+			serviceSpecMap[service.Name] = fillServiceSpec(service.Name, url, np, service.Weight, namespace, http)
+		}
+	} else {
+		if service.Tls != "" {
```

Comment:
it would be more readable if ALWAYS http is assumed, unless TLS is set

Reply:
```
serviceSpecMap[service.Name].URL = append(serviceSpecMap[service.Name].URL, fmt.Sprintf(http, url))
if service.Tls != "" {
  serviceSpecMap[service.Name].URL = append(serviceSpecMap[service.Name].URL, fmt.Sprintf(https, url))
}
```

This will be not slower or less-performant... and might even be optimized away. it makes the code more readable...

Reply:
Agree with this comment.

Reply:
also not sure what append actually does here... does it ADD or just format something?

maybe it is the structure... but it feels unclear to me.

Reply:
It appends, so this part of the code executes if the service is already present in serviceSecMap, then we just append the new route and weight to the existing values..

Reply:
But that will make it show two routes, suppose the service does not have insecure route? In that case also it will have the http route.. while actually it was not defined..

Reply:
might be a good thing to keep the full route in the `ServiceWeight` struct.. name needs to be changed, in that case we can avoid the check for the `Tls` field...